components/BinList.js

```diff
@@ -13,55 +13,37 @@ const BinList = () => {
   const navigation = useNavigation(); 
 
   useEffect(() => {
-    const fetchAllData = async () => {
-      try {
-        const dbRef = ref(database, 'trash-bin-database');
-        onValue(dbRef, (snapshot) => {
-          if (snapshot.exists()) {
-            const data = snapshot.val();
-            const bins = Object.keys(data).map(key => ({ id: key, ...data[key] }));
-            setBinData(bins);
-          } else {
-            setError('No data found');
-          }
-        }, {
-          onlyOnce: false, 
-        });
-      } catch (error) {
-        setError('Something went wrong!');
-      } finally {
-        setLoading(false);
-      }
-    };
-
-    fetchAllData();
-
     const dbRef = ref(database, 'trash-bin-database');
-    const unsubscribe = onValue(dbRef, (snapshot) => {
+    const handleData = (snapshot) => {
       if (snapshot.exists()) {
         const data = snapshot.val();
         const bins = Object.keys(data).map(key => ({ id: key, ...data[key] }));
         setBinData(bins);
+        setError(null);
       } else {
         setError('No data found');
       }
-    });
+      setLoading(false);
+    };
+
+    const unsubscribe = onValue(dbRef, handleData);
 
     return () => {
-      off(dbRef); 
+      unsubscribe(); 
```

Comment:
no start loading just end loading only. should be only call function in useEffec